README.md

```diff
@@ -1,160 +1,178 @@
-- `BITCOIN_RPC_PASS`
-- `BTC_WALLET_PASS`
-    
-and replace the IP with your Bitcoin node IP:
-- `BITCOIN_RPC_HOST`
-- `ZMQ_RAWBLOCK_URL`
-- `ZMQ_RAWTX_URL`
-
-**Configure for Babylon**
+The key env vars to set are the server IP addresses where your Bitcoin Signet node and Babylon finality system are deployed. Note that the ports are preconfigured in the Docker Compose files used for this deployment.
 
-set your Babylon key's mnemonic, the address should have some BBN tokens, it will be used in the following steps
+- `BITCOIN_RPC_HOST`: the Bitcoin Signet node's IP address.
+- `ZMQ_RAWBLOCK_URL`: the Bitcoin Signet node's IP address.
+- `ZMQ_RAWTX_URL`: the Bitcoin Signet node's IP address.
+- `CONSUMER_EOTS_MANAGER_ADDRESS`: the Babylon finality system's IP address.
+- `FINALITY_GADGET_RPC`: the Babylon finality system's IP address.
+- `NEXT_PUBLIC_FINALITY_GADGET_API_URL`: the Babylon finality system's IP address.
 
-- `BABYLON_PREFUNDED_KEY_MNEMONIC`
+Besides these, you will need to set the following variables:
 
-set the following variables to register your OP Stack chain to Babylon:
+- `BABYLON_PREFUNDED_KEY_MNEMONIC`: the mnemonic for the wallet you used to claim BBN tokens in the previous step.
+- `CONSUMER_ID`: this is the identifier for your OP-Stack chain registration on Babylon, you can set it to anything you want (the convention we use is `<chain_type>-<chain_name>-<chain_id>-<version>`, e.g. `op-stack-tohma-706114-0001`).
+- `CONSUMER_CHAIN_NAME`: this is a human-readable name for your chain.
+- `OP_FP_MONIKER`: this is a human-readable name for your OP-Stack chain's finality provider.
+- `L2_RPC_URL`: this is your OP-Stack chain's RPC URL.
 
-- `CONSUMER_ID`
-- `CONSUMER_CHAIN_NAME`
+### 3. Set Babylon keys
 
-set your OP Stack chain's finality provider moniker
+This step imports the pre-funded Babylon key, which will be used to deploy the different components of the Babylon finality system:
 
-- `OP_FP_MONIKER`
+- CW contract: a CosmWasm smart contract deployed on BabylonChain that enables / disables the finality gadget and stores finality votes (i.e. provides DA) for your OP-stack chain.
+- BTC Staker: a peripheral program that creates the BTC delegation for your consumer FP, and communicates with Babylon node.
+- Consumer EOTS Manager: a service that manages generation and submission of EOTS signatures for your consumer FP.
+- Consumer Finality Provider: a service that manages consumer FPs and submits finality votes to the CW contract.
 
-replace the IP with the Babylon finality system you deployed server IP:
+The prefunded Babylon account is also used for key actions such as:
 
-- `CONSUMER_EOTS_MANAGER_ADDRESS`
-- `FINALITY_GADGET_RPC`
+1. Registering your OP-Stack consumer chain on Babylon.
+2. Generates a new account for your OP-Stack chain's finality provider and funding it with BBN tokens to pay for gas when submitting finality votes.
```

Comment:
generating FP key also happens in this step. we should say sth like:

This step imports the pre-funded Babylon key, which will be used to in a few steps later (e.g. deploy cw contract, stake btc, register FP)

This step also creates a new key for FP which will be used later. This key is for FP's operational use

Reply:
fixed 

<img width="1425" alt="image" src="https://github.com/user-attachments/assets/6ff146dc-9ed3-4c26-8155-d77dcf8294e3">